Add testing with molecule and refactor code

brucellino opened the pull request:
Major changes:

  - Add molecule with default scenario
    - test coverage only includes CEntOS7
    - test coverage only includes default file existence
  - Refactor tasks:
    - Include OS-specific variables before OS-specific tasks
    - Change syntax of task inclusion from include: to include_tasks:
    - Change failure message condition to check if the release version
      exists for the OS

Commit:
Add testing with molecule and refactor code

Major changes:

  - Add molecule with default scenario
    - test coverage only includes CEntOS7
    - test coverage only includes default file existence
  - Refactor tasks:
    - Include OS-specific variables before OS-specific tasks
    - Change syntax of task inclusion from include: to include_tasks:
    - Change failure message condition to check if the release version
      exists for the OS

Commit:
Only test CEntOS7 in the default scenario

Commit:
Add stubs of UMD3 and UMD4 scenarios

Commit:
Test style compliance with inspec profile

Commit:
Give the license to EGI

Commit:
Update umd4 scenario

Commit:
Add UMD test matrix and use fashion police profile from supermarket

Commit:
Add default umd version for default scenario testing

Commit:
Improve test coverage

The UMD4 (default) scenario now includes test coverage for relevant
operating systems:

  - centos6 (sl6)
  - centos7 (sl7)
  - ubuntu trusty

Commit:
Fix build matrix

Commit:
add test role var to global

Commit:
It really helps to read the frikkin docs

Commit:
update vars for repos

orviz commented on vars/Debian9.yml:
```diff
@@ -0,0 +1,5 @@
+release_url:
+  3: http://repository.egi.eu/sw/production/umd/3/debian/dists/squeeze/main/binary-amd64/umd-release-3.0.0_all.deb
+sources_dir: /etc/apt/sources.list.d
+nick: squeeze
```
nick is not good

orviz commented on vars/Debian8.yml:
```diff
@@ -0,0 +1,5 @@
+release_url:
+  3: http://repository.egi.eu/sw/production/umd/3/debian/dists/squeeze/main/binary-amd64/umd-release-3.0.0_all.deb
+sources_dir: /etc/apt/sources.list.d
+nick: squeeze
```
nick is not good

orviz commented on vars/Debian7.yml:
```diff
@@ -0,0 +1,5 @@
+release_url:
+  3: http://repository.egi.eu/sw/production/umd/3/debian/dists/squeeze/main/binary-amd64/umd-release-3.0.0_all.deb
+sources_dir: /etc/apt/sources.list.d
+nick: squeeze
```
nick is not good

orviz reviewed (changes requested):
re-check nick names

Commit:
Change name of Debian release supported

Commit:
Don't use test, but converge->verify

Commit:
Use explicit controls

Commit:
Verbosity on ruby version

gwarf commented on .travis.yml:
```diff
@@ -21,6 +21,7 @@ install:
 - pip install molecule docker
 - gem install inspec
 before_script:
+- ruby --versio:
```
Doesn't look right.

brucellino replied:
good catch! thanks.

Commit:
Update gems before installing inspec

Commit:
... and that's how you know it's time to go home£

Commit:
Add required gems to the environment, dufus

Commit:
Make controls an array

Commit:
add repo slug

Commit:
Fix linting errorS

Commit:
general cleanup of README and add Travis badge

Commit:
Add ca_verification to molecule playbook

Commit:
Fix linting errors

Commit:
UMD3 is deprecated

Commit:
Add UMD4 testing to matrix

Commit:
Use more idempotent language in task names

Commit:
Use more idempotent language in task names

Commit:
Remove spurious task for UMD/Debian

Commit:
Use new role name

Commit:
Add baseline task to ensure some necessary packages are present

Commit:
Add slack hooks

Commit:
enable testing in Python 3

Commit:
Add repo tests

Commit:
add task to ensure that yum repo dir is present

Commit:
Add crl and repo tests

Commit:
remove slack hooks

Commit:
fix linting errors

Commit:
Well, that was easy.

Fixed #4
HT https://stackoverflow.com/questions/31019854/typeerror-cant-use-a-string-pattern-on-a-bytes-like-object-in-re-findall

Commit:
Patch travis with @brucellino's version

brucellino committed:
Merge branch 'master' into master

Commit:
Update ca package and add tests

brucellino committed:
Merge pull request #3 from brucellino/master

v0.1.0 RC

Commit:
name the platforms in testing consistently

Commit:
Push on green

brucellino committed:
Merge branch 'master' into master

Commit:
Remove secure variables from .travisl and put them in the build

Commit:
Merge branch 'master' of https://github.com/brucellino/ansible-role-umd

brucellino committed:
Merge pull request #5 from brucellino/master

Push on Green

Commit:
Add proper badge and fix after_success

brucellino committed:
Merge branch 'master' into master

brucellino committed:
Merge pull request #6 from brucellino/master

Add proper badge and fix after_success

Commit:
add container badge

Commit:
add a bit about installing the role

Commit:
Verbosity on Travis variables

Commit:
Proper logic on push

Commit:
Correct name of the image

Commit:
Add black panther inspec profile

Close #9
Close #10

Commit:
Fix name of the inspec command for black panther

Commit:
Update testing strategy and bump ca release to 1-95

Commit:
Remove black panther profile

Commit:
fix travis script

Commit:
Add pydist-xdist for parallel test execution

Commit:
Add pydist xdist for parallel testing

brucellino committed:
Merge pull request #3 from EGI-Foundation/cd

continuous delivery

brucellino committed:
Merge pull request #7 from EGI-Foundation/cd

continuous delivery of Quay images

Commit:
Remove words about the external dependency on epel repo

Commit:
Merge branch 'cd' of https://github.com/brucellino/ansible-role-umd into cd

Commit:
Merge branch 'master' of https://github.com/EGI-Foundation/ansible-role-umd

Commit:
Fix Galaxy warnings

    E702:
    Use 'galaxy_tags' rather than 'categories'
    E701:
    Role info should contain company

Commit:
Fix galaxy errors ( E206:
Variables should have spaces after {{ and before }}

brucellino committed:
Merge pull request #8 from brucellino/master

improve galaxy score

Commit:
Add package vulnerability tests

Commit:
Add ruamel.yaml requirement

brucellino committed:
Merge branch 'master' into master

Commit:
Remove test since Travis hides tokens

Commit:
Merge branch 'master' of https://github.com/brucellino/ansible-role-umd

orviz committed:
Fix: set minimum Ansible version to 2.4

Usage of include_tasks requires ansible>=2.4

Commit:
Expect the version of ntpdate to be >= the patched one

brucellino committed:
Merge pull request #11 from orviz/fix_version

Fix: set minimum Ansible version to 2.4

brucellino committed:
Merge branch 'master' into master
